app/javascript/src/ActiveDocs/OAS3Autocomplete.ts

```diff
@@ -100,7 +114,7 @@ export interface Response extends SwaggerUIResponse {
   text: string;
 }
 
-const autocompleteOAS3 = async (response: SwaggerUIResponse, accountDataUrl: string, serviceEndpoint: string): Promise<Response> => {
+export const autocompleteOAS3 = async (response: SwaggerUIResponse, accountDataUrl: string, serviceEndpoint: string): Promise<Response> => {
```

Comment:
> It seems this is being exported only for tests. IMO this has became a private function now, so it shouldn't be exported. Tests shouldn't condition the code, they should adapt to the code.

That is right, and I agree with you that ideally the tests should adapt to the code, and not the other way around. On the other hand, when we code, we generally think about making the code testable (splitting codes in smaller units, adding dependency injection mechanism to allow for easier mocking of the dependencies, etc.).

In this case, I would really prefer to test the function `autocompleteOAS3` itself, and not the only "public" function `autocompleteRequestInterceptor`.

I am not sure if there is another _easy_ way to do this, without exporting the function (there are ways, but not too straightforward).

> In this case, the tests to assert `autocompleteOAS3` logic should be included as tests for the **'fetching OpenAPI spec'** context of `autocompleteRequestInterceptor` test, since that's the 'unit' being exported.

I am not sure I understand what you mean :upside_down_face: 

@josemigallas what do you think?

Reply:
I meant basically move `'should inject servers to the spec'` and `'should autocomplete fields of OpenAPI spec with x-data-threescale-name property'` to line 100 after `'should update the response interceptor'`. But OK.

Reply:
@jlledom  like this? https://github.com/3scale/porta/pull/3103/commits/98ff758d69a9f501d0736ceb6a5f5b19292a3496